ambry-network/src/main/java/com.github.ambry.network/NetworkMetrics.java

```diff
@@ -118,20 +133,52 @@ public NetworkMetrics(MetricRegistry registry) {
     sslHandshakeCount = registry.counter(MetricRegistry.name(Selector.class, "SslHandshakeCount"));
     sslHandshakeErrorCount = registry.counter(MetricRegistry.name(Selector.class, "SslHandshakeErrorCount"));
     sslRenegotiationCount = registry.counter(MetricRegistry.name(Selector.class, "SslRenegotiationCount"));
+
+    connectionsDisconnectedCount =
+        registry.meter(MetricRegistry.name(NetworkClient.class, "ConnectionsDisconnectedCount"));
+    networkClientSendAndPollTime =
+        registry.histogram(MetricRegistry.name(NetworkClient.class, "NetworkClientSendAndPollTime"));
+    requestQueueTime = registry.histogram(MetricRegistry.name(NetworkClient.class, "RequestQueueTime"));
+    requestSendTime = registry.histogram(MetricRegistry.name(NetworkClient.class, "RequestSendTime"));
+    requestSendTotalTime = registry.histogram(MetricRegistry.name(NetworkClient.class, "RequestSendTotalTime"));
+    requestResponseRoundTripTime =
+        registry.histogram(MetricRegistry.name(NetworkClient.class, "RequestResponseRoundTripTime"));
+    requestResponseTotalTime = registry.histogram(MetricRegistry.name(NetworkClient.class, "RequestResponseTotalTime"));
+    connectionTimeOutError = registry.counter(MetricRegistry.name(NetworkClient.class, "ConnectionTimeOutError"));
+    networkClientIOError = registry.counter(MetricRegistry.name(NetworkClient.class, "NetworkClientIOError"));
+
     selectorNodeMetricMap = new HashMap<String, SelectorNodeMetric>();
   }
 
   /**
-   * Initializes a few network metrics for the selector
-   * @param activeConnections count of current active connections
+   * Registers the number of active connections for a selector
+   * @param numActiveConnections count of current active connections
+   */
+  void registerSelectorActiveConnections(final AtomicLong numActiveConnections) {
+    Gauge<Long> selectorActiveConnectionsCount = new Gauge<Long>() {
```

Comment:
I think your original idea of grouping this together is better - there is not much insight in having selector level granularity for this metric. Will follow up with you.